packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/settings.dart

```diff
@@ -51,13 +54,34 @@ class Settings {
   /// Web only.
   final bool ignoreUndefinedProperties;
 
+  /// Forces the SDK’s underlying network transport (WebChannel) to use long-polling.
+  ///
+  /// Each response from the backend will be closed immediately after the backend sends data
+  /// (by default responses are kept open in case the backend has more data to send).
+  /// This avoids incompatibility issues with certain proxies, antivirus software, etc.
+  /// that incorrectly buffer traffic indefinitely.
+  /// Use of this option will cause some performance degradation though.
+  final bool? webExperimentalForceLongPolling;
+
+  ///	Configures the SDK's underlying transport (WebChannel) to automatically detect if long-polling should be used.
+  ///
+  ///This is very similar to [webExperimentalForceLongPolling], but only uses long-polling if required.
+  final bool? webExperimentalAutoDetectLongPolling;
+
+  final WebExperimentalLongPollingOptions? webExperimentalLongPollingOptions;
```

Comment:
Missing a documentation here

Options that configure the SDK’s underlying network transport (WebChannel) when long-polling is used.These options are only used if experimentalForceLongPolling is true or if experimentalAutoDetectLongPolling is true and the auto-detection determined that long-polling was needed. Otherwise, these options have no effect.